soc/arm/st_stm32/stm32g0/soc.h

```diff
@@ -52,6 +48,9 @@
 #include <stm32g0xx_ll_usart.h>
 #endif
 
+/* Add include for DTS generated information */
+#include <generated_dts_board.h>
```

Comment:
why do we include this at the end of the file here?

Reply:
This is standard for ARM SoCs. We first include vendor headers, followed by including DTS generated headers.

Reply:
In this way we prevent from having inclusion cycles.

Reply:
this seems broken by design TBH
What does generated_dts_board.h pull in which requires it at the end of file and not the beginning?
And since nothing in the header uses the definitions, why is it pulled in here at all? Pulling in headers from other headers which aren't actually needed by the including header is what causes inclusion cycle problems

Reply:
Ok, I was  not clear sorry. It can be anywhere in the file. The inclusions were caused by including kernel headers here. That's what I removed. 
Just for consistency with other SOCs, I prefer to pull in the dts generated headers after the vendor headers but this should not matter.